### PR TITLE
Adding _ch to channel tag and _pl to planet tag in S5 variable names

### LINKED ISSUE #643: [Enhancement]: Add pl# and ch# to parameter names when fitting multiple planets and/or channels
### Instrument

Light curve fitting (Stages 4-6)

### What happened?

The current naming scheme is confusing when fitting multiple planets over multiple channels.

While we're at it, we should either:

1. Update the damped oscillator parameter names to avoid duplicate keys, or
2. Rewrite the code in `s6_Spectra.py` to only add parameter names if they are a perfect match.

### Error traceback output

_No response_

### What operating system are you using?

_No response_

### What version of Python are you running?

_No response_

### What Python packages do you have installed?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR reformats the fitted variable names in Stage 5 to use _ch# for different channels and _pl# for different planets. This will make the code far more robust since we can far more easily identify the channel and planet number, and this should also help with Stage 6 bugs. It will also make the variable names a bit more human readable, and this PR should also fix multiple bugs related to trying to manually set different priors or fixed values for different channels.

These changes resolve #643

While I was at it, I also added support for ecosw and esinw in astrophysical models which avoids the well known ecc>0 bias of fitting for ecc and w directly.

I still need to fully integrate and test the _pl tags, especially for starry, and I still need to do a bunch of testing. For now I'm just opening a draft PR to help me keep track of my progress, so don't worry about reviewing it until I mark it as ready for review.

Note, this change is going to break backwards compatibility for S6 running on old S5 outputs, but that isn't a huge deal since S6 isn't critical and we're working towards v1.0

### CODE REVIEW

**Comment by taylorbell57** [comment]
Alright, this PR is now known to work well through to Stage 5, but I haven't yet tested to see what impact this has on Stage 6. My suspicion is that Stage 6 will merge together the spectra of all planets, so I'll likely need to address that

**Comment by kevin218** [comment]
@taylorbell57  Can you make sure `split_channels.py` is included in this PR?  It looks like `get_trim` was added, but I don't see it.

**Comment by taylorbell57** [comment]
@kevin218, not sure what you mean - `get_trim()` and split_channels.py are both present (see below). I don't think `get_trim` is new to this PR either, but I could be wrong
https://github.com/kevin218/Eureka/blob/caf582374ff9881f20538fc2a8019fa5388ee20e/src/eureka/lib/split_channels.py#L4-L28

**Comment by kevin218** [comment]
I thought it was new. My bad.

File: src/eureka/S5_lightcurve_fitting/models/BatmanModels.py [hunk 1]
```diff
@@ -94,6 +110,33 @@ def __init__(self, model, pid=0, channel=0):
             if model.parameters.dict[item0][1] == 'free':
                 item0 += self.channel_id
             setattr(self, 'a', model.parameters.dict[item0][0])
+        # Allow for (ecc, w) or (ecosw, esinw)
+        if (self.ecosw is None) and ('ecc' in model.parameters.dict.keys()):
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S5_lightcurve_fitting/models/BatmanModels.py:114]: The code is fast (just one more if/else statement and simple transformation) and is comparable to other choices like ensuring both `rp` and `rprs` are supported

File: src/eureka/S6_planet_spectra/s6_spectra.py [hunk 2]
```diff
@@ -216,79 +242,101 @@ def plot_spectra(eventlabel, ecf_path=None, s5_meta=None, input_meta=None):
...
-                        meta.y_label = r'$R_{\rm p}/R_{\rm *}$'
-                    elif meta.y_param == 'fp' or meta.y_param == 'fpfs':
+                        suffix = getPlanetSuffix(meta)+getChannelSuffix(meta)
+                        meta.y_label = '$R_{\\rm p'+suffix+'}/R_{\\rm *}$'
+                    elif meta.y_param_basic in ['fp', 'fpfs']:
                         # Eclipse depth
-                        meta.y_label = r'$F_{\rm p,day}/F_{\rm *}$'
-                    elif meta.y_param == 'fn':
+                        suffix = getPlanetSuffix(meta)+getChannelSuffix(meta)
+                        meta.y_label = '$F_{\\rm p,day'+suffix+'}/F_{\\rm *}$'
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S6_planet_spectra/s6_spectra.py:263]: I think I agree that `F_{p#,day}/F_*` makes the most sense

**Review by kevin218 — DISMISSED** [review]
Check for Flake8 issues

**Review by kevin218 — APPROVED** [review]

**Pull request merged**